grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -834,16 +834,16 @@ class RequisitionService {
 
     void triggerRequisitionStatusTransition(Requisition requisition, User currentUser) {
         // OBPIH-5134 Request approval feature implements additional status transitions for a request
-        // If fulfilling location does not require approval we omit all other status transitions and set it to VERIFYING
         switch(requisition.status) {
             case RequisitionStatus.PENDING_APPROVAL:
                 if (requisition.origin.approvalRequired) {
                     transitionRequisitionStatus(requisition, RequisitionStatus.PENDING_APPROVAL, EventCode.PENDING_APPROVAL, currentUser)
                     requisition.approvalRequired = true
-                } else {
-                    transitionRequisitionStatus(requisition, RequisitionStatus.VERIFYING, EventCode.SUBMITTED, currentUser)
-                    requisition.approvalRequired = false
+                    break
```

Comment:
now - don't we want to send an event at all if we set it for `VERIFYING`? (when we send a request when origin doesn't support approvals)?

Reply:
what do you mean?
the `break` is inside of an `if()` and if `requisition.origin.approvalRequired` is `false` it will go to next case which is `case RequisitionStatus.VERIFYING:`.

I wanted to avoid duplicate code with this approach but if you feel it is too confusing then let me know, I'll either add a comment or duplicate it for the sake of clarity.

Reply:
yes, it will go there, but from the frontend you still always send `PENDING_APPROVAL` and if a location doesn't support the approvals, it won't run the `transitionRequisitionStatus` method at all. Or am I missing something?

Reply:
it will run `triggerRequisitionStatusTransition` and set the status to `VERIFYING`.
Anyways, these state transitions are getting a bit too confusing, I think I need to rethink my approach to which status to send and how to react to it.